libs/frontend/ui-build/src/components/component-menu/ComponentStyles.vue

```diff
@@ -92,6 +94,27 @@ const addStyle = () => {
   createStyle()
 }
 
+const setValueWrap = (oldEntry: IInheritedStyleEntry, newValue: string) => {
+  // If the property was inherited, create a new entry so undo works as expected
+  // Otherwise, the style would be reverted to `oldEntry` on undo
+  if (oldEntry.sourceType !== StyleSourceType.Custom) {
+    createStyle({ prop: oldEntry.property, value: newValue })
+  } else {
+    setValue(oldEntry, newValue)
+  }
+}
```

Comment:
@abemscac This is my alternate solution, I haven't fully tested it but it seems to work. The idea is that if we correctly track style changes here by making sure the inherited style isn't used as `oldStyle` in the command, undo will work as expeced.